### PR TITLE
354 Create unit test Vignette

### PR DESCRIPTION
**Pull request by millerg23**

### CODE REVIEW

File: vignettes/unit_test_guidance.Rmd [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+## Why write unit tests?
```
Comment (thomas-neitmann) [inline vignettes/unit_test_guidance.Rmd:21]: Please use title case: **U**nit **T**ests.

File: vignettes/unit_test_guidance.Rmd [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+simple. If tests are hard to read, developers are more likely to misunderstand 
+them and introduce bugs. Test cases could be used as a form of documentation, so
+they obviously need to be readable.
+
+### Running Unit Tests Part of the Build Process
+Automate the whole process of running the unit tests and taking some action when
+they fail. Your build process should execute your unit tests and mark the build 
+as broken when the tests fail.
+
+# Writing Unit Tests in Admiral
```
Comment (thomas-neitmann) [inline vignettes/unit_test_guidance.Rmd:79]: Please use {admiral} throughout the vignette.

File: vignettes/unit_test_guidance.Rmd [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+Ensure you give a meaningful explanation of the test in the testthat call, as 
+these will be compiled in the package validation report. Having the name of the
+function and test ID included in title will also help with tracability.
+
+Once you have tested your unit test program, you can run all unit tests from
+the console, as follows.
+
+```
+devtools::check
+```
```
Comment (thomas-neitmann) [inline vignettes/unit_test_guidance.Rmd:136]: Please replace with `devtools::test()`

File: vignettes/unit_test_guidance.Rmd [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+## Automation of Unit Tests
+
+When a user actions a pull request in admiral github repo, the unit tests are 
+automatically run and pull request will be denied if any unit tests fail.
```
Comment (thomas-neitmann) [inline vignettes/unit_test_guidance.Rmd:141]: Please write as **G**it**H**ub everywhere.

File: vignettes/unit_test_guidance.Rmd [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+This is only needed in cases where the complexity and level of flexibility of 
+your function justifies it, e.g. see the test script: https://github.com/Roche-GSK/admiral/blob/master/tests/testthat/test-derive_extreme_flag.R.
+
+## Exported Functions
+Don’t forget to add a unit test for each exported function.
+
+## Set up the Test Script
+Within {admiral} folder https://github.com/Roche-GSK/admiral/tree/master/tests/testthat, 
+add a script with the naming convention “test-<function name>.R” and use the 
+following format:
```
Comment (thomas-neitmann) [inline vignettes/unit_test_guidance.Rmd:109]: I would slightly reoword. The test file name should have the same name as the implementation file. If there are multiple function in `some_funs.R` file then these should all be tested in `test-some_funs.R`.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Great work 👏 

Just some minor comments.

File: vignettes/unit_test_guidance.Rmd (hunk 1 above)
Comment (millerg23) [reply vignettes/unit_test_guidance.Rmd]: Done

File: vignettes/unit_test_guidance.Rmd (hunk 2 above)
Comment (millerg23) [reply vignettes/unit_test_guidance.Rmd]: Done.

File: vignettes/unit_test_guidance.Rmd (hunk 3 above)
Comment (millerg23) [reply vignettes/unit_test_guidance.Rmd]: Done

File: vignettes/unit_test_guidance.Rmd (hunk 4 above)
Comment (millerg23) [reply vignettes/unit_test_guidance.Rmd]: Done

File: vignettes/unit_test_guidance.Rmd (hunk 5 above)
Comment (millerg23) [reply vignettes/unit_test_guidance.Rmd]: Done, added the extra piece about creating the unit test file from console.

File: vignettes/unit_test_guidance.Rmd [hunk 6]
```diff
@@ -105,14 +105,39 @@ Don’t forget to add a unit test for each exported function.
...
+  input <- tibble::tribble(
+    ~inputvar1, ~inputvar2, ...
+    <Add Test Data Scenarios>
+    ...
+  )
+
+  expected_output <- mutate(input, outputvar = c(<Add Expected Outputs>))
 
-context("test-<function name>")
+  expect_equal(<function name>(input), expected_output)
```
Comment (thomas-neitmann) [inline vignettes/unit_test_guidance.Rmd:126]: When comparing **datasets** we use `expect_dfs_equal()`. That should explicitly called out.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: vignettes/unit_test_guidance.Rmd (hunk 6 above)
Comment (millerg23) [reply vignettes/unit_test_guidance.Rmd:126]: I have added a note stating this, let me know if you needed more detail added.

**Review by thomas-neitmann — APPROVED** [review]
Thanks!

**Pull request merged**